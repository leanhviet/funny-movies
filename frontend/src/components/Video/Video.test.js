// Libs
import React from 'react'
import { shallow } from 'enzyme'

// Components
import { VideoComponent } from './index'

const props = {
  id: 'XMNRRE11',
  title: 'Khoa Pug and Vuong Pham',
  sharedBy: 'vietla@gmail.com',
  likeCount: 12222,
  dislikeCount: 10,
  description: 'Johnny Dang lua dao chiem doat 30 ty'
}

describe('Video component', () => {
  const wrapper = shallow(<VideoComponent />)
  const wrapperWithProps = shallow(<VideoComponent {...props} />)

  it('should be defined Video component', () => {
    expect(wrapper).toBeDefined()
  })

  // Testing Snapshot
  it('should be render snapshot Video component', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapperWithProps).toMatchSnapshot()
  })

  // Testing default props
  it('should be render defaultProps Video component', () => {
    expect(VideoComponent.defaultProps.id).toEqual('')
    expect(VideoComponent.defaultProps.title).toEqual('')
    expect(VideoComponent.defaultProps.sharedBy).toEqual('')
    expect(VideoComponent.defaultProps.likeCount).toEqual(0)
    expect(VideoComponent.defaultProps.dislikeCount).toEqual(0)
    expect(VideoComponent.defaultProps.description).toEqual('')
  })

  // Testing mock props
  it('should be render mockProps Input component', () => {
    expect(wrapperWithProps.find('iframe').props().src).toContain(
      props.id,
    )
    expect(wrapperWithProps.find('iframe').props().title).toEqual(
      props.title,
    )
    expect(wrapperWithProps.find('CardText').props().children).toEqual(
      props.description,
    )
  })
})
