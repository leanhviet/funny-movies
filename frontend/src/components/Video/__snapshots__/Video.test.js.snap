// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video component should be render snapshot Video component 1`] = `
<Card
  body={false}
  className="video-cmp"
>
  <Row>
    <Col
      className="video-cmp__col1"
      md={6}
    >
      <iframe
        allow="autoplay; fullscreen"
        className="video-cmp__col1__iframe"
        frameBorder="0"
        src="https://www.youtube.com/embed/?rel=0"
        title=""
      />
    </Col>
    <Col
      className="video-cmp__col2"
      md={6}
    >
      <CardBody>
        <CardTitle />
        <p>
          Shared by: 
        </p>
        <div>
          <span>
            0
             
            <i
              aria-hidden="true"
              className="fa fa-thumbs-o-up"
              style={
                Object {
                  "marginRight": "20px",
                }
              }
            />
          </span>
          <span>
            0
             
            <i
              aria-hidden="true"
              className="fa fa-thumbs-o-down"
            />
          </span>
        </div>
        <p>
          Description:
        </p>
        <CardText
          className="video-cmp__col2__desc"
        />
      </CardBody>
    </Col>
  </Row>
</Card>
`;

exports[`Video component should be render snapshot Video component 2`] = `
<Card
  body={false}
  className="video-cmp"
>
  <Row>
    <Col
      className="video-cmp__col1"
      md={6}
    >
      <iframe
        allow="autoplay; fullscreen"
        className="video-cmp__col1__iframe"
        frameBorder="0"
        src="https://www.youtube.com/embed/XMNRRE11?rel=0"
        title="Khoa Pug and Vuong Pham"
      />
    </Col>
    <Col
      className="video-cmp__col2"
      md={6}
    >
      <CardBody>
        <CardTitle>
          Khoa Pug and Vuong Pham
        </CardTitle>
        <p>
          Shared by: 
          vietla@gmail.com
        </p>
        <div>
          <span>
            12222
             
            <i
              aria-hidden="true"
              className="fa fa-thumbs-o-up"
              style={
                Object {
                  "marginRight": "20px",
                }
              }
            />
          </span>
          <span>
            10
             
            <i
              aria-hidden="true"
              className="fa fa-thumbs-o-down"
            />
          </span>
        </div>
        <p>
          Description:
        </p>
        <CardText
          className="video-cmp__col2__desc"
        >
          Johnny Dang lua dao chiem doat 30 ty
        </CardText>
      </CardBody>
    </Col>
  </Row>
</Card>
`;
